monkey/common/di_container.py

```diff
@@ -56,12 +61,45 @@ def register_instance(self, interface: Type[T], instance: T):
         self._instance_registry[interface] = instance
         DIContainer._del_key(self._type_registry, interface)
 
+    def register_convention(self, type_: Type[T], name: str, instance: T):
+        """
+        Register an instance as a convention
+
+        At times — particularly when dealing with primative types — it can be useful to define a
+        convention for how dependencies should be resolved. For example, you might want any class
```

Comment:
```suggestion
        At times — particularly when dealing with primative types — it can be useful to resolve dependencies by parameter name. For example, you might want any class
```

Reply:
It's not resolved by parameter name, though. It's resolved by the combination of type and name. It's a convention.